applications/substrate-evm-adapter.md

```diff
@@ -0,0 +1,189 @@
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | ETH-RPC Adapter | We will create a generic ETH-RPC Adapter service for Substrate chains. It will support the most essential ETH RPC calls to make it work with the existing tools. Main goals here will be ability to connect to Metamask, `web3.js`, read Substrate chain's pallets' state and support subscriptions. Adapter will have option of running local `smoldot` instance or connecting to remote RPC node. |
+| 2. | Deno module | We will create a Deno module that can connect to Substrate chain as an ETH-RPC adapter + light client. |
+| 3. | Javascript package | We will provide a javascript package that can connect to Substrate chain as an ETH-RPC adapter + light client. |
+| 4. | Unit tests | We will provide comprehensive unit tests for the adapter. |
+
+### Milestone 2 — EVM Adapter Pallet
+
+- **Estimated duration:** 9 weeks
+- **FTE:**  1
+- **Costs:** 18,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | MIT |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| **0e.** | Article | I will publish an article that explains the complete lifecycle and future plans of the project |
+| 1. | EVM Adapter Pallet | We will create a pallet that will be responsible for bridging the gap between the ETH RPC adapter and the Substrate chain. It will have the ability to dispatch FRAME calls, opt-in possibility to execute EVM bytecode, and handle account mapping. Another main responsibility of the pallet will be handling signature verification. Some parts of this pallet can be inspired from `frontier`'s pallet `pallet-ethereum` but needs refinement and some modification. |
+| 2. | Substrate Node | We will create a Substrate node that has two runtimes: with and without the `pallet-evm`. Both will contain `evm-adapter` pallet, but only the one with `pallet-evm` will be able to execute EVM bytecode. This will demonstrate two main use-cases of this pallet. |
```

Comment:
How is this Node different from existing EVM chains like Moonbeam?

Reply:
moonbeam and many other EVM compatible chains built with Frontier have the ETH RPC baked in in the client. They also add a lot pallets, runtime APIs and other changes to the runtime to ensure full EVM compatibility. Integrated ETH RPC complicates the code, really [slows]((https://github.com/polkadot-evm/frontier/issues/1067#issue-1731136963)) down the node, tps takes a hit and makes it hard to benchmark etc.

This deliverable in comparison, will have the most minimal setup: normal Substrate client and runtime with stripped down, lightweight version of the `pallet-evm` and `pallet-evm-adapter`. `rpc-adapter` is a standalone light client and does not have the problems of integrated rpc